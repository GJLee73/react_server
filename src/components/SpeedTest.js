import React from 'react';
import {connect} from 'react-redux';
import {getSpeedRequest} from 'actions/speedTest';

class SpeedTest extends React.Component{
	constructor(props){
		super(props);
		//this.setState({});
		this.componentDidMount = this.componentDidMount.bind(this);
	}

	componentDidMount(){
		this.props.getSpeedRequest().then(
			() => {
				if(this.props.status == 'SUCCESS'){
					Materialize.toast('SUCCESS');
					this.setState(this.props.data);
				}
				else{
					Materialize.toast('Failure');

				}
			}
		)
	}

	render(){
		return(
			<div>
				{this.props.status}
				<form>
            		<input type='BUTTON' defaultValue="networkTest" onClick={this.componentDidMount}/>
            	</form>
            	{this.props.data? this.props.data.client.ip : undefined}
            </div>
		)
	}
}

const mapStateToProps = (state) =>{
    return{
        status: state.speedTest.status,
        data: state.speedTest.data
    };
};

const mapDispatchtoProps = (dispatch) => {
    return {
        getSpeedRequest: ()=>{
            return dispatch(getSpeedRequest());
        }
    };
}

export default connect(mapStateToProps, mapDispatchtoProps)(SpeedTest);