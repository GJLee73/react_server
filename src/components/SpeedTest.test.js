import React from 'react';
import {shallow} from 'enzyme'
import SpeedTest from './SpeedTest'

describe('SpeedTest test',()=>{
    var getSpeed = jest.fn(()=>Promise.resolve())
    var getSpeed_fail = jest.fn(()=>Promise.reject())
    var wrapper = shallow(<SpeedTest data={{client:undefined}} getSpeed={getSpeed}/>)
  
    describe('snapshot test',()=>{
        it('client가 없을때',()=>{
            wrapper.setProps({status:'init',data:{client:undefined}})
            
            expect(wrapper).toMatchSnapshot()
        })
        it('client가 있을때',()=>{
            wrapper.setProps({data:{client:{ip:'125.0.0.1'},speeds:{download:'100',upload:'100'},server:{ping:'10'}}})
            expect(wrapper).toMatchSnapshot()
        })
    })
    it('click test',()=>{
        wrapper.find('input').simulate('click')
        expect(getSpeed.mock.calls.length).toBe(1)
    })
    it('click test when return reject' ,()=>{
        wrapper.setProps({getSpeed:getSpeed_fail})
        wrapper.find('input').simulate('click')
        expect(getSpeed_fail.mock.calls.length).toBe(1)
    })
})