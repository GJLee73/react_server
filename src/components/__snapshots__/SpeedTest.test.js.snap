// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeedTest test snapshot test client가 없을때 1`] = `
<div>
  init
  <form>
    <input
      defaultValue="networkTest"
      onClick={[Function]}
      type="BUTTON"
    />
  </form>
</div>
`;

exports[`SpeedTest test snapshot test client가 있을때 1`] = `
<div>
  init
  <form>
    <input
      defaultValue="networkTest"
      onClick={[Function]}
      type="BUTTON"
    />
  </form>
  <ul>
    <li>
      ip: 
      125.0.0.1
    </li>
    <li>
      download: 
      100
    </li>
    <li>
      upload: 
      100
    </li>
    <li>
      ping: 
      10
    </li>
  </ul>
</div>
`;
